apps/pipelines/nodes/nodes.py

```diff
@@ -284,11 +299,15 @@ class LLMResponseWithPrompt(LLMResponse, HistoryMixin):
         description="Custom actions to enable for the bot",
         json_schema_extra=UiSchema(widget=Widgets.multiselect, options_source=OptionsSource.custom_actions),
     )
-
-    built_in_tools: list[str] = Field(
+    built_in_tools: list[BuiltInTools] = Field(
         default_factory=list,
         description="Built in tools provided by the LLM model",
-        json_schema_extra=UiSchema(widget=Widgets.multiselect, options_source=OptionsSource.built_in_tools),
+        json_schema_extra=UiSchema(widget=Widgets.built_in_tools, options_source=OptionsSource.built_in_tools),
+    )
+    tool_config: ToolConfigModel | None = Field(
+        default_factory=dict,
+        description="Configuration for builtin tools",
+        json_schema_extra=UiSchema(widget=Widgets.none),
     )
```

Comment:
This should be a dict of dicts to allow for configuration from multiple tools. The keys in this dict would be the values of `BuiltInTools`

```suggestion
    tool_config: dict[str, ToolConfigModel] = Field(
        default_factory=dict,
        description="Configuration for builtin tools",
        json_schema_extra=UiSchema(widget=Widgets.none),
    )
```

Reply:
done